project/frontend/src/components/LocationFinder.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+
+class LocationFinder extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      userInput: '',
+      locationName: '',
+      submitted: false,
+    };
+    this.changeState = this.changeState.bind(this);
+    this.getLocationFromGeolocate = this.getLocationFromGeolocate.bind(this);
+    this.getLocationFromText = this.getLocationFromText.bind(this);
+  }
+
+  changeState(event) {
+    this.setState({ [event.target.name]: event.target.value });
+  }
+
+  /** 
+   * Uses Geolocation to get the user's current coordinates.
+   * Gets formatted address to display from Geocoder.
+   */
+  getLocationFromGeolocate() {
+    if (navigator.geolocation) {
+      const google = window.google;
+      const geocoder = new google.maps.Geocoder();
+      navigator.geolocation.getCurrentPosition((position) => {
+        const currLocation = { lat: position.coords.latitude, lng: position.coords.longitude };
+        geocoder.geocode({ location: currLocation }, (results, status) => {
+          if (status === 'OK') {
+            const locationName = results[0].formatted_address;
+            this.setState({ locationName });
+            this.setState({ submitted: true });
+            this.props.sendData({ currLocation, locationName });
+          } else {
+            alert('Geocode was not successful for the following reason: ' + status);
+          }
+        });
+      });
+    } else {
+      alert('Cannot find your location');
```

Comment:
i'm not even sure if we're keeping this alert or if it's a filler right now but if we are could we tell them to try typing in their location?

Reply:
Good idea! I think we should eventually remove the alerts altogether but for now it would still be helpful to have an alert with more useful information than what I have here lol